src/components/MapboxMap/index.tsx

```diff
@@ -78,13 +78,20 @@ const MapboxMap: React.FC<MapboxMapProps> = ({
       setMap(newMap);
     });
     newMap.on("render", () => {
-      const center = newMap.getCenter();
-      setTransform({
-        zoom: newMap.getZoom(),
-        center: [center.lng, center.lat],
-        bearing: newMap.getBearing(),
-        pitch: newMap.getPitch(),
-      });
+      const newCenter = newMap.getCenter();
+      const zoom = newMap.getZoom();
+      const center: [number, number] = [newCenter.lng, newCenter.lat];
+      const bearing = newMap.getBearing();
+      const pitch = newMap.getPitch();
+      // no need to update the state if the values are the same
+      if (zoom !== transform?.zoom && center !== transform?.center && bearing !== transform?.bearing && pitch !== transform?.pitch) {
```

Comment:
Doesn't `[0, 0] !== [0, 0]` evaluate to true? Probably best to use lodash's `isEqual` for the center comparison.

Reply:
Good call thank you

Reply:
see e96c0b48913e44f7c6c92ec06b2123cb08cf332c